apps/files_external/lib/Lib/Storage/FtpConnection.php

```diff
@@ -171,7 +171,7 @@ private function parseUnixItem(string $item, string $directory): array {
 		];
 	}
 
-	private function normalizePermissions(string $permissions) {
+	private function normalizePermissions(string $permissions): mixed {
```

Comment:
```suggestion
	private function normalizePermissions(string $permissions): string {
```
But we might need to update the code for psalm to trust us on this, unless psalm has a stub for array_reduce with templates and stuff.